src/providers/twitch/api/Helix.cpp

```diff
@@ -1991,6 +2004,157 @@ void Helix::sendWhisper(
         .execute();
 }
 
+QString Helix::formatHelixUserListErrorString(
+    QString userType,
+    HelixUserListError error,
+    QString message
+) {
+    using Error = HelixUserListError;
+
+    QString errorMessage = QString("Failed to get list of ") + userType + QString(": ");
+
+    switch (error)
+    {
+        case Error::Forwarded: {
+            errorMessage += message;
+        }
+        break;
+
+        case Error::UserMissingScope: {
+            errorMessage += "Missing required scope. "
+                            "Re-login with your "
+                            "account and try again.";
+        }
+        break;
+
+        case Error::UserNotAuthorized: {
+            errorMessage += "You don't have permission to "
+                            "perform that action.";
+        }
+        break;
+
+        case Error::Unknown: {
+            errorMessage += "An unknown error has occurred.";
+        }
+        break;
+    }
+    return errorMessage;
+}
+
+// Recursive function with a maximun page of 50
+void Helix::getApiListRecursive(
+    HelixUserList *list,
```

Comment:
Let's avoid a pointer here. You could pass it by reference, pass a `shared_ptr`, or move it from call to call.

Reply:
Moved reference to list to the parent function referenced in a callback.